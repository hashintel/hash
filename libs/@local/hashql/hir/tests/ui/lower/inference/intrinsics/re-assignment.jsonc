//@ run: pass
//@ description: Tests type inference when an intrinsic function is assigned to a variable, which is then reassigned to another variable, and finally the second variable is referenced.
["let", "foo", "::core::math::add", ["let", "bar", "foo", "bar"]]
