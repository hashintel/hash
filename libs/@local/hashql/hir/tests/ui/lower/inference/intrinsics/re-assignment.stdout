════ Initial HIR ═══════════════════════════════════════════════════════════════

#let foo:0 = ::core::math::add in
#let bar:0 = ::core::math::add in
bar:0

════ HIR after type inference ══════════════════════════════════════════════════

#let foo:0 = ::core::math::add in
#let bar:0 = ::core::math::add in
bar:0

════ Intrinsics ════════════════════════════════════════════════════════════════

┌─ ::core::math::add
└→ ::core::math::add

┌─ ::core::math::add
└→ ::core::math::add

┌─ bar:0
└→ ::core::math::add

